### PR TITLE
Implemented basic FavoritesFragment layout and functionality (swipe e…

### PR DESCRIPTION
**Pull request by elswenson**
…ither direction to remove from favorites), few minor layout changes to MenuFragment layouts

### CODE REVIEW

File: GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
+package edu.ucsb.cs.cs190i.gauchogrub.gauchogrub;
```
Comment (echistyakov) [inline GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java]: Let's organize our src root folder, it's getting a little crowded with all kinds of files.

└ ui
&nbsp;&nbsp;&nbsp;&nbsp;└ activities
&nbsp;&nbsp;&nbsp;&nbsp;└ fragments
&nbsp;&nbsp;&nbsp;&nbsp;└ adapters

Comment (elswenson) [inline GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java]: It definitely is, I looked into it at some point and had some imports flip out, so I didn't do it then. I'll take a look again now.

File: GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    private String diningCommon;
+    private int diningCommonId;
+    private Context context;
+    private FavoriteRecyclerAdapter thisAdapter;
+
+    protected FavoriteRecyclerAdapter(String diningCommon, Context context) {
+        super(MenuItem.$TYPE);
+        dataStore = ((GGApp) context.getApplicationContext()).getData();
+        this.diningCommon = diningCommon;
+        this.diningCommonId = dataStore.select(DiningCommon.class).where(DiningCommon.NAME.eq(diningCommon)).get().first().getId();
```
Comment (echistyakov) [inline GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java]: Each segment in long queries like this one should go on a new line for readability.
`dataStore`
`.select(...)`
`.where(...)`
`.get(...)`
`.first(...)`

File: GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        this.thisAdapter = this;
+        this.context = context;
+    }
+
+
+    @Override
+    public Result<MenuItem> performQuery() {
+        return dataStore.select(MenuItem.class).join(Favorite.class).on(MenuItem.ID.eq(Favorite.MENU_ITEM_ID))
```
Comment (echistyakov) [inline GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java]: Same as above.

File: GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        // Remove reduntant (v), (vgn), or (w/ nuts) from name
+        String NUTS_STRING = context.getString(R.string.parsable_has_nuts);
+        String VEG_STRING = context.getString(R.string.parsable_veg);
+        String VGN_STRING = context.getString(R.string.parsable_vgn);
+        viewHolder.favoriteTextView.setText(menuItem.getTitle()
+                .replace(NUTS_STRING, "")
+                .replace(VEG_STRING, "")
+                .replace(VGN_STRING, ""));
```
Comment (echistyakov) [inline GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/FavoriteRecyclerAdapter.java]: `trim()` to remove any white space on the sides?

**Pull request merged**